components/board/customlinux/config.go

```diff
@@ -0,0 +1,53 @@
+package customlinux
+
+import (
+	"os"
+
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/board/genericlinux"
+)
+
+// GenericLinuxPin describes a gpio pin on a linux board.
+type GenericLinuxPin struct {
+	Name            string `json:"name"`
+	Ngpio           int    `json:"ngpio"`
+	RelativeID      int    `json:"relative_id"`
+	PWMChipSysFSDir string `json:"pwm_chip_dir,omitempty"`
+	PWMID           string `json:"pwm_id,omitempty"`
+}
+
+// GenericLinuxPins describes a list of pins on a linux board.
+type GenericLinuxPins struct {
+	Pins []GenericLinuxPin `json:"pins"`
+}
+
+// Validate ensures all parts of the config are valid.
+func (config *GenericLinuxPin) Validate(path string) error {
+	if config.Name == "" {
+		return utils.NewConfigValidationFieldRequiredError(path, "name")
+	}
+
+	if config.PWMChipSysFSDir != "" && config.PWMID == "" {
+		return utils.NewConfigValidationFieldRequiredError(path, "pwm_id")
+	}
+	return nil
```

Comment:
Discussion question: should we make sure that the relative ID is at least 0 and less than the ngpio value? Part of me says "yes, that's what it's supposed to be" and part of me says "no, people know what it's supposed to be and if they're doing weird things it's probably on purpose." What do other folks think?